mirrord/layer/tests/issue2614.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg(target_os = "linux")]
+#![feature(assert_matches)]
+#![warn(clippy::indexing_slicing)]
+
+use std::{os::unix::fs::PermissionsExt, path::Path, time::Duration};
+
+use rand::{thread_rng, Rng};
+use rstest::rstest;
+
+mod common;
+
+pub use common::*;
+
+/// Verify that issue [#2614](https://github.com/metalbear-co/mirrord/issues/2614) is fixed.
```

Comment:
Maybe this comment could include a sentence about the issue just so you don't have to check the link? I know its pretty obvious from the test but might be nice
```suggestion
/// Verify that issue [#2614](https://github.com/metalbear-co/mirrord/issues/2614) is fixed
/// and the file open mode is honoured on bypass.
```